### PR TITLE
Style Guide: Passing around too many parameters, #22805

### PR DESCRIPTION
**Pull request by patriknw**
Refs #22805

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/style-guide.md [hunk 1]
```diff
@@ -124,3 +124,69 @@ Some reasons why you may want to use the functional style:
 
 @@@
 
+## Passing around to many parameters
+
+On thing you will quickly run into when using the functional style is that you need to pass around many parameters.
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/style-guide.md]: one

File: akka-docs/src/main/paradox/typed/style-guide.md [hunk 2]
```diff
@@ -124,3 +124,69 @@ Some reasons why you may want to use the functional style:
...
+
+Let's add `name` parameter and timers to the previous `Counter` example. A first approach would be to just add those
+as parameters separate parameters:
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/style-guide.md]: parameters twice

File: akka-docs/src/main/paradox/typed/style-guide.md [hunk 3]
```diff
@@ -124,3 +124,69 @@ Some reasons why you may want to use the functional style:
...
+The "constructor" parameters can be @scala[immutable]@java[`final`] instance fields and can be accessed from
+member methods.
+
+Scala
+:  @@snip [StyleGuideDocExamples.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala) { #fun-style-setup-params3 }
+
+Java
+:  @@snip [StyleGuideDocExamples.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/StyleGuideDocExamples.java) { #fun-style-setup-params3 }
+
+That's nice. One thing to cautious with here is that it's important that you create a new instance for
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/style-guide.md]: to _be_ cautious?

File: akka-docs/src/main/paradox/typed/style-guide.md [hunk 4]
```diff
@@ -124,3 +124,69 @@ Some reasons why you may want to use the functional style:
...
+each spawned actor, since those parameters must not be shared between different actor instances. That comes natural
+when creating the instance from `Behaviors.setup` as in the above example. Having a
+@scala[`apply` factory method in the companion object and making the constructor private is recommended.]
+@java[static `create` factory method and making the constructor private is highly recommended.]
+
+This can also be useful when testing the behavior by creating a test subclass that overrides certain methods in the
+class. The test would create the instance without the @scala[`apply` factory method`]@java[static `create` factory method].
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/style-guide.md]: runaway backtick

**Review by johanandren — COMMENTED** [review]
Looks good, language comments

**Comment by patriknw** [comment]
thx, fixed

**Review by johanandren — APPROVED** [review]

File: akka-docs/src/main/paradox/typed/style-guide.md [hunk 5]
```diff
@@ -124,3 +124,69 @@ Some reasons why you may want to use the functional style:
 
 @@@
 
+## Passing around to many parameters
```
Comment (chbatey) [inline akka-docs/src/main/paradox/typed/style-guide.md]: /to/too/

**Review by chbatey — COMMENTED** [review]
Nice

**Review by chbatey — APPROVED** [review]

**Review by raboof — APPROVED** [review]

File: akka-docs/src/main/paradox/typed/style-guide.md (hunk 5 above)
Comment (patriknw) [reply akka-docs/src/main/paradox/typed/style-guide.md]: thx, fixed

**Pull request merged**